syz-verifier/exectask.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/google/syzkaller/pkg/rpctype"
+	"github.com/google/syzkaller/prog"
+)
+
+type EnvDescr int64
+
+const (
+	AnyEnvironment EnvDescr = iota
+	NewEnvironment
+	// TODO: add CleanVMEnvironment support.
+
+	EnvironmentsCount
+)
+
+// ExecTask is the atomic analysis entity. Once executed, it could trigger the
+// pipeline propagation fof the program.
+type ExecTask struct {
+	CreationTime   time.Time
+	Program        *prog.Prog
+	ID             int64
+	ExecResultChan ExecResultChan
+
+	priority int // The priority of the item in the queue.
+	// The index is needed by update and is maintained by the heap.Interface methods.
+	index int // The index of the item in the heap.
```

Comment:
I see it's only assigned, but I don't see where it's ever used. Why is it needed?

Reply:
fixed